lib/apiv2/profile.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiv2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	KindProfile     = "Profile"
+	KindProfileList = "ProfileList"
+)
+
+// Profile contains the details a security profile resource.  A profile is set of security rules
+// to apply on an endpoint.  An endpoint (either a host endpoint or an endpoint on a workload) can
+// reference zero or more profiles.  The profile rules are applied directly to the endpoint *after*
+// the selector-based security policy has been applied, and in the order the profiles are declared on the
+// endpoint.
+type Profile struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	Metadata metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the Profile.
+	Spec ProfileSpec `json:"spec,omitempty"`
+}
+
+// ProfileSpec contains the specification for a security Profile resource.
+type ProfileSpec struct {
```

Comment:
Now that I think about it, we should probably have a `LabelsToApply` spec field here in order to be specific and to model kubernetes behavior.

Having the `Metadata.Labels` field be the ones that are inherited feels wrong and scary.

Reply:
Sounds good.